### PR TITLE
Fix index deletion

### PR DESCRIPTION
**Pull request by Toflar**
See https://github.com/contao/contao/issues/140

### CODE REVIEW

**Review by leofeyer — CHANGES_REQUESTED** [review]
Unfortunately, the fix does not work yet. It will add the `DROP INDEX` command twice, once in the `ALTER_TABLE` section and again in the `ALTER_CHANGE` section, which leads to a `Can't DROP INDEX …; check that it exists` error.

@m-vo discovered that the Doctrine schema manager already sorts the queries to take care of the problem we are trying to fix here, so maybe we should just drop our logic and use the pre-sorted queries?

File: installation-bundle/src/Database/Installer.php [hunk 1]
```diff
@@ -235,7 +235,11 @@ private function checkEngineAndCollation(array &$sql, Schema $toSchema): void
             if ($deleteIndexes) {
                 $platform = $this->connection->getDatabasePlatform();
 
-                foreach ($table->getIndexes() as $index) {
+                if (!$fromSchema->hasTable($tableName)) {
+                    continue;
+                }
+
+                foreach ($fromSchema->getTable($tableName)->getIndexes() as $index) {
                     if ('primary' !== $index->getName()) {
```
Comment (leofeyer) [inline installation-bundle/src/Database/Installer.php:243]: This needs to be `strtolower($index->getName())` to fix https://github.com/contao/contao/issues/140#issuecomment-436284620.

**Comment by leofeyer** [comment]
I have fixed this intermediately in 3d9c5943f6cd933a0aad539d5ffddca7c55981af, however we should definitely review this and use the Doctrine routines if possible IMHO.

**Comment by m-vo** [comment]
You may want to have a look if #169 also fixes this problem. There I'll just kept the original sorting order - in my use case to allow dropping tables with foreign keys (drop key first, then table) which doctrine already supports.

**Comment by leofeyer** [comment]
Ok, then I'm closing this in favor of #169.

**Pull request closed without merging**